targets/gameboy-advance-mb.ld

```diff
@@ -0,0 +1,84 @@
+OUTPUT_ARCH(arm)
+ENTRY(_start)
+
+/* Note: iwram is reduced by 96 bytes because the last part of that RAM
+ * (starting at 0x03007FA0) is used for interrupt handling.
+ */
+MEMORY {
+    ewram   : ORIGIN = 0x02000000, LENGTH = 256K   /* on-board work RAM (2 wait states) */
+    iwram   : ORIGIN = 0x03000000, LENGTH = 32K-96 /* in-chip work RAM (faster) */
+}
+
+__stack_size_irq = 1K;
+__stack_size_usr = 2K;
+
+SECTIONS
+{
+    .text :
+    {
+        KEEP (*(.init))
+        *(.text)
+        . = ALIGN(4);
+    } >ewram
+
+    .rodata :
+    {
+        . = ALIGN(4);
+        *(.rodata)
+        *(.rodata*)
+        . = ALIGN(4);
+    } >ewram
+
+    /* Put the stack at the bottom of RAM, so that the application will
+     * crash on stack overflow instead of silently corrupting memory.
+     * See: http://blog.japaric.io/stack-overflow-protection/ */
+    .stack (NOLOAD) :
+    {
+        . = ALIGN(4);
+        _stack_top_irq = .;
+        . += __stack_size_irq;
+        _stack_top = .;
+        . += __stack_size_usr;
+    } >iwram
+
+    /* Start address (in flash) of .data, used by startup code. */
+    _sidata = LOADADDR(.data);
+
+    /* Globals with initial value */
+    .data :
+    {
+        . = ALIGN(4);
+        _sdata = .;        /* used by startup code */
+        *(.data)
+        *(.data*)
+        *(.iwram .iwram.*)
+        . = ALIGN(4);
+        _edata = .;        /* used by startup code */
+    } >iwram AT>ewram
```

Comment:
This confuses me. Normally you would copy the `.data` section from flash to RAM, and the `>iwram AT>ewram` part tells the linker to put the area at ewram but relocate it as if it was at iwram. In this case, everything resides in RAM so there is no need to copy the `.data` section. Therefore, I would suggest to remove `_sidata` as it is no longer necessary (if it is, it is still copying the `.data` section even though it shouldn't) and replacing `>iwram AT>ewram` with just `>iwram`.